internal/util/update/update_test.go

```diff
@@ -1499,6 +1541,446 @@ func TestCommand_Run_subpackages(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "updated setters on root package",
+			initialUpstream: pkgbuilder.NewRootPkg().
+				WithKptfile(
+					pkgbuilder.NewKptfile().
+						WithSetters(
+							pkgbuilder.NewSetter("my-setter", "foo"),
+						),
+				),
+			updatedUpstream: testutil.Content{
+				Pkg: pkgbuilder.NewRootPkg().
+					WithKptfile(
+						pkgbuilder.NewKptfile().
+							WithSetters(
+								pkgbuilder.NewSetter("my-setter", "bar"),
+							),
+					),
+			},
+			expectedResults: []resultForStrategy{
+				{
+					strategies: []StrategyType{FastForward, KResourceMerge},
+					expectedLocal: pkgbuilder.NewRootPkg().
+						WithKptfile(
+							pkgbuilder.NewKptfile().
+								WithUpstreamRef("upstream", "/", "master").
+								WithSetters(
+									pkgbuilder.NewSetter("my-setter", "bar"),
+								),
+						),
+				},
+			},
+		},
+		{
+			name: "subpackages are updated based on the version specified in the parent Kptfile",
+			refRepos: map[string][]testutil.Content{
+				"foo": {
+					{
+						Pkg: pkgbuilder.NewRootPkg().
+							WithKptfile().
+							WithResource(pkgbuilder.DeploymentResource),
+						Branch: "master",
+					},
+					{
+						Pkg: pkgbuilder.NewRootPkg().
+							WithKptfile().
+							WithResource(pkgbuilder.ConfigMapResource),
+						Tag: "v1.0",
+					},
+				},
+			},
+			initialUpstream: pkgbuilder.NewRootPkg().
+				WithKptfile(
+					pkgbuilder.NewKptfile().
+						WithSubpackages(
+							pkgbuilder.NewSubpackage("foo", "/", "master", "fast-forward", "foo"),
+						),
+				),
+			updatedUpstream: testutil.Content{
+				Pkg: pkgbuilder.NewRootPkg().
+					WithKptfile(
+						pkgbuilder.NewKptfile().
+							WithSubpackages( // In the new version of the parent Kptfile, v1.0 of the subpackage should be used.
+								pkgbuilder.NewSubpackage("foo", "/", "v1.0", "fast-forward", "foo"),
+							),
+					).
+					WithResource(pkgbuilder.DeploymentResource),
+			},
+			expectedResults: []resultForStrategy{
+				{
+					strategies: []StrategyType{FastForward, KResourceMerge},
+					expectedLocal: pkgbuilder.NewRootPkg().
+						WithKptfile(
+							pkgbuilder.NewKptfile().
+								WithUpstreamRef("upstream", "/", "master").
+								WithSubpackages(
+									pkgbuilder.NewSubpackage("foo", "/", "v1.0", "fast-forward", "foo"),
+								),
+						).
+						WithResource(pkgbuilder.DeploymentResource).
+						WithSubPackages(
+							pkgbuilder.NewSubPkg("foo").
+								WithKptfile(
+									pkgbuilder.NewKptfile().
+										WithUpstreamRef("foo", "/", "v1.0"),
+								).
+								WithResource(pkgbuilder.ConfigMapResource),
+						),
+				},
+			},
+		},
+		{
+			name: "subpackage with changes can not be updated with fast-forward strategy",
+			refRepos: map[string][]testutil.Content{
```

Comment:
This need not go in this PR but renaming this to `remotePkgRepos` makes it more readable.